Frameworks/Foundation/NSProgress.mm

```diff
@@ -252,10 +278,22 @@ - (int64_t)completedUnitCount {
 /**
  @Status Interoperable
 */
-- (double)fractionCompleted {
-    // Override synthesized getter so that this can be calculated dynamically
+- (void)setTotalUnitCount:(int64_t)inUnitCount {
+    @synchronized(self) { // Property is atomic
+        double ratio = (double)inUnitCount / _totalUnitCount;
+        _totalUnitCount = inUnitCount;
+        [self _updateCompletedUnitsBy:0
+                  fractionCompletedBy:(_fractionCompleted / ratio) - _fractionCompleted
```

Comment:
I can't actually. I need to update _totalUnitCount in the middle to make the call to _updateCompletedUnits work.

---
In reply to: [95283538](https://github.com/Microsoft/WinObjC/pull/1663#discussion_r95283538) [](ancestors = 95283538)